pkg/templates/pdbminavailable/template.go

```diff
@@ -0,0 +1,173 @@
+	"golang.stackrox.io/kube-linter/pkg/templates/pdbminavailable/internal/params"
+	pdbV1 "k8s.io/api/policy/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	templateKey = "pdb-min-available"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Pod Disruption Budget Configuration Checks",
+		Key:         templateKey,
+		Description: "Flag non-optimal PDB configurations",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.PodDisruptionBudget},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return minAvailableCheck, nil
+		}),
+	})
+}
+
+func minAvailableCheck(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+
+	var results []diagnostic.Diagnostic
+
+	// Get the PDB provided
+	pdb, ok := object.K8sObject.(*pdbV1.PodDisruptionBudget)
+	if !ok {
+		return nil
+	}
+
+	// If MinAvailable isn't set, then no need to check
+	if pdb.Spec.MinAvailable == nil {
+		return results
+	}
+
+	// Extract the MinAvailable value from the spec and check if it's a number or percentage
+	value, isPercent, err := getIntOrPercentValueSafelyFromString(pdb.Spec.MinAvailable.String())
+	if err != nil {
+		return []diagnostic.Diagnostic{
+			{
+				Message: fmt.Sprintf("PDB has invalid MinAvailable value: %v", err),
+			},
+		}
+	}
+
+	// If the value is a percentage, handle the case where the MinValue is set to 100%
+	// as DeploymentLike's replica counts don't need to be compared
+	if isPercent && value == 100 {
+		return []diagnostic.Diagnostic{
+			{
+				Message: "PDB has minimum available replicas set to 100 percent of replicas",
+			},
+		}
+	}
+
+	// Build the label selector for the PDB to use for comparison
+	labelSelector, err := metaV1.LabelSelectorAsSelector(&metaV1.LabelSelector{
+		MatchLabels:      pdb.Spec.Selector.MatchLabels,
+		MatchExpressions: pdb.Spec.Selector.MatchExpressions},
+	)
+
+	if err != nil {
+		return []diagnostic.Diagnostic{
+			{
+				Message: fmt.Sprintf("PDB has invalid label selector: %v", err),
+			},
+		}
+	}
+
+	// Evaluate Deploymet Likes in the lintContext to see if they have MinAvailable set too low
+	deploymentLikes, err := getDeploymentLikeObjects(lintCtx, labelSelector, pdb.Namespace)
+	if err != nil {
+		return []diagnostic.Diagnostic{
+			{
+				Message: fmt.Sprintf("Failed to retrieve deployments matching the PDB's label selector within namespace %s: %v", pdb.Namespace, err.Error()),
+			},
+		}
+	}
+
+	for _, dl := range deploymentLikes {
+		pdbMinAvailable := value
+		replicas, _ := extract.Replicas(dl)
+		if isPercent {
+			// Calulate the actual value of the MinAvailable with respect to the Replica count if a percentage is set
+			pdbMinAvailable = int(math.Floor(float64(replicas) * (float64(value) / float64(100))))
+		}
```

Comment:
Is this necessary? 

IIUC, the only case when `pdbMinAvailable` would be lower than or equal to `replicas` is for `value == 100` which is handled separately above. I think that we have to loop through the deployment-like objects only when `minAvailable` is set to an absolute value.

Reply:
Sure, I guess we can return here.

Reply:
It is necessary indeed. Kubernetes rounds maxUnavailable and minAvailable **up**, **not  down**.

See
* [PDB docs](https://kubernetes.io/docs/tasks/run-application/configure-pdb/#rounding-logic-when-specifying-percentages)
* [Related PDB code](https://github.com/kubernetes/kubernetes/blob/23be9587a0f8677eb8091464098881df939c44a9/pkg/controller/disruption/disruption.go#L539)
* [GetValueFromIntOrPercent implementation](https://github.com/kubernetes/apimachinery/blob/9cb1f71d9bce60a0ec671b70613892c055f8ff49/pkg/util/intstr/intstr.go)
```suggestion
		if isPercent {
			// Calulate the actual value of the MinAvailable with respect to the Replica count if a percentage is set
			pdbMinAvailable = int(math.Ceil(float64(replicas) * (float64(value) / float64(100))))
		}
```